### ISSUE TITLE
After removing one ComponentRenderer column the other renderer stop working

### ISSUE THREAD

**Issue by Telefonmann1024** [datenhahn/componentrenderer#26]
Hi,
in my project I get an IllegalStateException after removing a column and scroll some lines down.
To reproduce this error I take the ClassicGridTab class from the demo and insert a Button to remove a column during runtime just before addComponent(grid) in line 182:

``` java
addComponent(new Button("Remove " + GENERATED_DELETE + " Column", new ClickListener() {

    @Override
    public void buttonClick(ClickEvent event) {
        if (grid.getColumn(GENERATED_DELETE) != null)
            grid.removeColumn(GENERATED_DELETE);

    }
}));
```

After pushing the button all renderer stop working.

![componentrenderer](https://cloud.githubusercontent.com/assets/19590451/16444015/214b9b1e-3ddc-11e6-8430-5283251bf3f5.jpg)
I have test this with vaadin version 7.6.2 and 7.6.6 and with Firefox, Chrom and IExplorer.

And after scrolling down some line I get the same exception as in my project - maybe another issue:

```
java.lang.IllegalStateException: Renderer is not attached to any parent
        at com.vaadin.ui.Grid$AbstractGridExtension.getParentGrid(Grid.java:4366)
        at de.datenhahn.vaadin.componentrenderer.ComponentRenderer.generateData(ComponentRenderer.java:101)
        at com.vaadin.server.communication.data.RpcDataProviderExtension.getRowData(RpcDataProviderExtension.java:395)
        at com.vaadin.server.communication.data.RpcDataProviderExtension.pushRowData(RpcDataProviderExtension.java:383)
        at com.vaadin.server.communication.data.RpcDataProviderExtension.access$900(RpcDataProviderExtension.java:65)
        at com.vaadin.server.communication.data.RpcDataProviderExtension$2.requestRows(RpcDataProviderExtension.java:293)
```

**Comment by datenhahn** [comment]
bug was reproducable, but I have to have a look at the datagenerator code and what happens in Grid with datagenerator when a column is removed, how urgent is a fix for you (are you on a production project?)

**Comment by Telefonmann1024** [comment]
Yes, I'm working on a production project. I use the ComponentRenderer to show one sparkline in a grid (> 5000 rows) and most of my customers using firefox. So I have the issue 7 problem and looking for a workaround. Simple hiding the column has no effect. If there is a solution for issure 7, it would help me much more.

**Comment by datenhahn** [comment]
Ok, then I will hurry with the fix, can you post me the link to the issue you mean 7 is closed and doesn't seem right (memory)

**Comment by Telefonmann1024** [comment]
Sorry, I've looked at the issue tab and have confused the open issues (7) with the issue actual showing. 
My problem is issue #24 "Scrolling not working as intended".

**Comment by datenhahn** [comment]
I have a fix, but before releasing it I have to create the test , etc. I pushed a branch containing the fix, you could do me favor if you test it in your production environment.

Checkout the branch "columnremovalfix" (clone my repo and do "git checkout columnremovalfix") containing the fix and build it with:
mvn clean install

Then use this dependency in your project and rebuild it:

```
    <dependency>
        <groupId>de.datenhahn.vaadin</groupId>
        <artifactId>componentrenderer</artifactId>
        <version>1.0.0-columnremovalfix</version>
    </dependency>
```

**Comment by Telefonmann1024** [comment]
That's looks very good for me. I test it sucessfully with my project.
After removing the column no exception is thrown and the grid has it original speed.
Adding and removing the column multiple times has no negative effects.

Thank you - good work.

**Comment by datenhahn** [comment]
Released version 1.0.1 containing the fix:

```
<dependency>
   <groupId>de.datenhahn.vaadin</groupId>
   <artifactId>componentrenderer</artifactId>
   <version>1.0.1</version>
</dependency>
```
